### PR TITLE
Add docs for CREATE INDEX CONCURRENTLY

### PR DESCRIPTION
**Pull request by lnguyen-yugabyte**
Add docs for CREATE INDEX CONCURRENTLY as it is now supported (in #10799)

### CODE REVIEW

File: docs/content/preview/api/ysql/the-sql-language/statements/ddl_create_index.md [hunk 1]
```diff
@@ -56,7 +56,9 @@ The following table explains some of the differences between creating an index o
 | Parallelizes index loading? | yes | no |
 
 To enable online schema migration for one `CREATE INDEX`, use `CREATE INDEX CONCURRENTLY`.
+
```
Comment (deeps1991) [inline docs/content/preview/api/ysql/the-sql-language/statements/ddl_create_index.md:59]: Actually it feels a bit hard to understand which is the default behavior. I believe we are concurrent by default? Also I believe ysql_disable_index_backfill=true will take precedence over CREATE INDEX CONCURRENTLY? Would be good to mention that as well.

File: docs/content/stable/api/ysql/the-sql-language/statements/ddl_create_index.md [hunk 2]
```diff
@@ -74,22 +75,17 @@ on individually, which you should do in the following cases:
...
-Enable online schema migration (see [Semantics](#semantics) for details), with some restrictions:
-* Upon creating an index on a temporary table, online schema migration will be disabled.
-* `CREATE INDEX CONCURRENTLY` is not supported for partitioned table.
-* `CREATE INDEX CONCURRENTLY` is not supported inside a transaction block.
-
 ### NONCONCURRENTLY
 
 Disable online schema migration (see [Semantics](#semantics) for details).
 
+Concurrent is the default, but the grammar does not yet support `CONCURRENTLY`.
```
Comment (deeps1991) [inline docs/content/stable/api/ysql/the-sql-language/statements/ddl_create_index.md:87]: Maybe we can just remove this line? We already said we don't support CONCURRENTLY keyword at line 46.

Comment (lnguyen-yugabyte) [reply docs/content/stable/api/ysql/the-sql-language/statements/ddl_create_index.md:87]: It is removed in the latest version.

File: docs/content/preview/api/ysql/syntax_resources/the-sql-language/statements/create_index,index_elem.grammar.md [hunk 3]
```diff
@@ -1,5 +1,5 @@
 ```ebnf
-create_index ::= CREATE [ UNIQUE ] INDEX [ NONCONCURRENTLY ]  
+create_index ::= CREATE [ UNIQUE ] INDEX [ CONCURRENTLY | NONCONCURRENTLY ]
```
Comment (ddhodge) [inline docs/content/preview/api/ysql/syntax_resources/the-sql-language/statements/create_index,index_elem.grammar.md:2]: Diagram not updated - do you need to run the diagram generator rrdiagram.jar

File: docs/content/preview/api/ysql/the-sql-language/statements/ddl_create_index.md [hunk 4]
```diff
@@ -74,17 +75,22 @@ Creating an index on a partitioned table automatically creates a corresponding i
...
 * [Row-level geo-partitioning](../../../../../explore/multi-region-deployments/row-level-geo-partitioning/) is being used. In this case, create the index separately on each partition to customize the tablespace in which each index is created.
+* `CREATE INDEX CONCURRENTLY` is not supported for partitioned tables (see [CONCURRENTLY](#concurrently)).
 
 ### UNIQUE
 
 Enforce that duplicate values in a table are not allowed.
 
+### CONCURRENTLY
+Enable online schema migration (see [Semantics](#semantics) for details), with some restrictions:
+* Upon creating an index on a temporary table, online schema migration will be disabled.
```
Comment (ddhodge) [inline docs/content/preview/api/ysql/the-sql-language/statements/ddl_create_index.md:86]:
```suggestion
* When creating an index on a temporary table, online schema migration is disabled.
```

File: docs/content/preview/api/ysql/the-sql-language/statements/ddl_create_index.md [hunk 5]
```diff
@@ -74,17 +75,22 @@ Creating an index on a partitioned table automatically creates a corresponding i
...
+* `CREATE INDEX CONCURRENTLY` is not supported for partitioned table.
```
Comment (ddhodge) [inline docs/content/preview/api/ysql/the-sql-language/statements/ddl_create_index.md:87]:
```suggestion
* `CREATE INDEX CONCURRENTLY` is not supported for partitioned tables.
```

**Review by ddhodge — COMMENTED** [review]
minor comments.

**Comment by deeps1991** [comment]
@lnguyen-yugabyte Is it just me? I previously saw some changes to the "stable" version as well (ie. which does not support CONCURRENTLY option). But now those changes are no longer present. Why are they removed? Aren't they applicable to stable?

**Comment by lnguyen-yugabyte** [comment]
@deeps1991 : the change is not backported to a stable release (so only on master but not on 2.12). When we do a new stable release (2.14), I'll send a separate PR to update the docs accordingly for the stable release.

**Review by deeps1991 — APPROVED** [review]

**Pull request merged**